packages/web-app-files/src/mixins/actions/restore.ts

```diff
@@ -47,7 +57,88 @@ export default {
     ...mapMutations('runtime/spaces', ['UPDATE_SPACE_FIELD']),
     ...mapMutations(['SET_QUOTA']),
 
-    async $_restore_trigger({ resources }) {
+    getParentFolderFromResource(resource) {
+      return resource.path.slice(0, resource.path.lastIndexOf(resource.name))
+    },
+    getWebdavParentFolderFromResource(resource) {
+      const parentPath = this.getParentFolderFromResource(resource)
+      return buildWebDavFilesPath(this.user.id, parentPath)
+    },
+    async collectRestoreConflicts(resources) {
+      const parentFolders = {}
+      const conflicts = []
+      const resolvedResources = []
+      for (const resource of resources) {
+        const webDavParentPath = this.getWebdavParentFolderFromResource(resource)
+
+        // ? check if parent folder has already been requested
+        let parentResources = []
+        if (webDavParentPath in parentFolders) {
+          parentResources = parentFolders[webDavParentPath]
+        } else {
+          const listResponse = await this.$client.files.list(
+            webDavParentPath,
+            1,
+            DavProperties.Default
+          )
+          parentResources = listResponse.map((i) => buildResource(i))
+          parentFolders[webDavParentPath] = parentResources
+        }
+        // ? Check for naming conflict in parent folder and between resources batch
+        const hasConflict =
+          parentResources.some((e) => e.name === resource.name) ||
+          resources.filter((e) => e.id !== resource.id).some((e) => e.name === resource.name)
```

Comment:
I don't quite get this check. Imagine I have the following 2 files:

* `/someFile`
* `/folder/someFile`

I delete both `someFile`, now I want to restore both at the same time -> conflict dialog shows. Isn't the parent path check enough?